modules/ROOT/pages/indexes-for-nearest-neighbor.adoc

```diff
@@ -0,0 +1,431 @@
+The semantic meaning of the vector may itself dictate which similarity function to choose.
+Consult the documentation for the particular model used to generate vector embeddings.
+The documentation may suggest a preference for certain similarity functions.
+Otherwise, understanding the differences can help make an informed choice.
+
+.Similarity functions
+[%header,cols="d,m,e"]
+|===
+| Name | Case Insensitive Argument | Key similarity feature
+
+| xref:#indexes-vector-similarity-euclidean[Euclidean]
+| "euclidean"
+| distance
+
+| xref:indexes-vector-similarity-cosine[Cosine]
+| "cosine"
+| angle
+
+|===
+
+For {l2-norm}alized vectors, such that their length, image:l2norm_is_1.svg["The l2-norm of vector v equals 1"], both Euclidean and cosine similarity functions produce the same similarity ordering.
+
+[[indexes-vector-similarity-euclidean]]
+=== Euclidean similarity
+
+Euclidean is useful when the _distance_ between the vectors is what determines how similar two vectors are.
+
+A valid vector for a Euclidean vector index is when all vector components can be represented finitely in {ieee-754} single precision.
+
+Euclidean interprets the vectors in Cartesian coordinates.
+The measure is related to the Euclidean distance, how far two points are from one another.
+However, that is unbounded, and less useful as a similarity score.
+Euclidean similarity bounds the square of Euclidean distance.
+
+image::euclidean_similarity_equation.svg["The Euclidean of vector v and vector u is defined as 1 over the quantity 1 plus the square of the l2-norm of vector v subtract vector u, which exists in the bounded set of real numbers between 0 exclusive and 1 inclusive."]
+
+[[indexes-vector-similarity-cosine]]
+=== Cosine similarity
+
+Cosine should be used when the _angle_ between the vectors is what determines how similar two vectors are.
+
+A valid vector for a Cosine vector index is when:
+
+* All vector components can be represented finitely in {ieee-754} single precision.
+* And its {l2-norm} is non-zero and can be represented finitely in {ieee-754} single precision.
+
+Cosine interprets the vectors in Cartesian coordinates.
+The measure is related to the angle between the two vectors.
+However, an angle can be described in many units, sign conventions, and periods.
+The trigonometric cosine of this angle is both agnostic to the aforementioned and bounded.
+Cosine similarity rebounds the trigonometric cosine.
+
+image::cosine_similarity_equation.svg["The cosine of vector v and vector u is defined as half of the quanity 1 plus the scalar product of v hat u hat, which equals half of the quantity 1 plus the scalar product of vector v vector u over the product of the l2-norm of vector v and the l2 norm ov vector u, which exists in the bounded set of real numbers between 0 inclusive and 1 inclusive."]
+Where the trigonometric cosine is given by the scalar product of two unit vectors.
```

Comment:
What is this "Where" referring to? Could it be replaced with something like: "The trigonometric cosine in the above equation..."?

```suggestion
Where the trigonometric cosine is given by the scalar product of two unit vectors.
```

Reply:
_where_ is often used in academic writing following an equation specifically to clarify parts of it; often in lowercase as it's considered a continuation of the equation's sentence.

There is no standard function notation for the angle between two vectors in terms of those two vectors; thus cannot write something like `cos(angleBetween(v,u))`, and have to write `v.u` which is more common.

Sure, I wouldn't be opposed to something like
```suggestion
The trigonometric in the above equation is given by the scalar product of the two unit vectors.
```
Or, more preferably,
```suggestion
where scalar product of the two unit vectors is the trigonometric cosine of the angle between the two vectors.
```
or some hybrid. What are your thoughts and suggestions? Whichever we go with, I will make the same change in the operations manual which also has these.